File: src/main/java/org/flighthub/Repository/ITicketRepository.java
package org.flighthub.Repository;

import org.flighthub.Domain.Ticket;

import java.util.Optional;
import java.util.UUID;

public interface ITicketRepository extends Repository<UUID, Ticket>{
    Optional<Ticket> findOne(UUID id);

    Iterable<Ticket> findAll();

    Ticket save(Ticket entity);

    Optional<Ticket> delete(UUID id);

    Optional<Ticket> update(Ticket entity);
}

File: Server - C#/persistence/ITicketRepository.cs
using FlightHubC_.Domain;
using FlightHubC_.Repository;
using System.Collections.Generic;
using System;
namespace FlightHubC_.Repository
{

    public interface ITicketRepository : IRepository<Guid, Ticket>
    {
        Ticket FindOne(Guid id);
        IEnumerable<Ticket> FindAll();
        Ticket Save(Ticket entity);
        Ticket Delete(Guid id);
        Ticket Update(Ticket entity);
    }
}